payload/reggae/dub/interop/dublib.d

```diff
@@ -267,9 +267,32 @@ package struct Dub {
 // module that don't do that on purpose for speed reasons.
 auto fullDub(in string projectPath) @trusted {
     import dub.dub: DubClass = Dub;
+    import dub.packagemanager: PackageManager;
     import dub.internal.vibecompat.inet.path: NativePath;
 
-    auto dub = new DubClass(projectPath);
+    // Cache the PackageManager.
+    // A reggaefile.d with lots of dub{Package,Dependant} targets benefits from
+    // this, possibly depending on the size of the dub packages cache.
+    static class DubWithCachedPackageManager : DubClass {
+        this(string rootPath) {
+            super(rootPath);
+        }
+
+        override PackageManager makePackageManager() const {
+            static PackageManager cachedPM = null;
+            if (!cachedPM) {
+                // The base implementation creates a PackageManager using
+                // `<root pkg path>/.dub/packages` as one of its repositories.
+                // With this cache here, this means that we only support that
+                // dir for the first root project, and apply it to all other
+                // root projects.
+                cachedPM = super.makePackageManager();
```

Comment:
@Geod24: Any suggestions? Maybe constructing directly here and feeding it `NativePath.init` as root package path?

Reply:
@Geod24: Looking at https://github.com/dlang/dub/blob/0d4858eb473a31a058132c8692d499effdb1cb28/source/dub/packagemanager.d#L116-L124, it doesn't look like that special `<root pkg path>/.dub/packages/` repo can be disabled. Is there a way?

Reply:
The `PackageManager` is not designed to handle changing `Location`, and was originally designed with 3 hardcoded `Location`s, making any kind of extensibility hard (e.g. the `PlacementLocation` enum).
Caching the `PackageManager` is an option but seems the most hacky one. Another option is to only instantiate one `Dub` instance (might require some changes to Dub), or to make package loading lazy. Per and myself have made some progress in the lazy loading, I don't know the Reggae codebase well enough to say how much Dub needs to change to allow for what it does. So ATM I don't have a satisfying answer, sorry.

Reply:
Okay nevermind, I found a way, and am using the original reggae project directory now.